protocol/messenger_testing_utils.go

```diff
@@ -63,7 +63,10 @@ type MessengerSignalsHandlerMock struct {
 }
 
 func (m *MessengerSignalsHandlerMock) SendWakuFetchingBackupProgress(response *wakusync.WakuBackedUpDataResponse) {
-	m.wakuBackedUpDataResponseChan <- response
+	select {
+	case m.wakuBackedUpDataResponseChan <- response:
+	default:
+	}
```

Comment:
Can you please elaborate why changed to non-blocking?

Reply:
It blocked my test. To be honest, I don't fully know how this code works, I just discovered that if I change this code to this, it makes my test work and the other test that uses it still works, so it sounded fine.

Reply:
@osmaczko I found the problem, `wakuBackedUpDataResponseChan ` was never initialized in this test, so it was failing to write to a `nil` channel here.
Fixed:
https://github.com/status-im/status-go/blob/025de8f90031cfce3855455ca5e750dca0068318/protocol/messenger_backup_test.go#L266-L268

Reply:
It would still be nice to have a `select/case` here, but without context it's incorrect to have. In this case better to have the tests hanging(failing).